From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Istvan Zoltan Ballok <istvan.zoltan.ballok@sap.com>
Date: Fri, 29 Nov 2024 11:33:39 +0100
Subject: [PATCH 07/29] Store the objects in the objectsToGarbageCollect map

Previously, the map was used as a set for the object IDs: the object ID was
stored as a key in the map, and the value was an empty struct.

Now, we store the objects in the map as a value.

Later, we'll need the resource version and the UID fields to create the
`DeleteOptions` object with the preconditions. By storing the objects in the
map, we can avoid querying the API server for these objects again.
---
 .../controller/garbagecollector/reconciler.go                | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/pkg/resourcemanager/controller/garbagecollector/reconciler.go b/pkg/resourcemanager/controller/garbagecollector/reconciler.go
index 90144ef8e7..fc0ba69221 100644
--- a/pkg/resourcemanager/controller/garbagecollector/reconciler.go
+++ b/pkg/resourcemanager/controller/garbagecollector/reconciler.go
@@ -48,7 +48,7 @@ func (r *Reconciler) Reconcile(reconcileCtx context.Context, _ reconcile.Request
 
 	var (
 		labels                  = client.MatchingLabels{references.LabelKeyGarbageCollectable: references.LabelValueGarbageCollectable}
-		objectsToGarbageCollect = map[objectId]struct{}{}
+		objectsToGarbageCollect = map[objectId]*metav1.PartialObjectMetadata{}
 	)
 
 	for _, resource := range []struct {
@@ -65,12 +65,13 @@ func (r *Reconciler) Reconcile(reconcileCtx context.Context, _ reconcile.Request
 		}
 
 		for _, obj := range objList.Items {
+			obj.SetGroupVersionKind(corev1.SchemeGroupVersion.WithKind(resource.kind))
 			if obj.CreationTimestamp.Add(*r.MinimumObjectLifetime).UTC().After(r.Clock.Now().UTC()) {
 				// Do not consider recently created objects for garbage collection.
 				continue
 			}
 
-			objectsToGarbageCollect[objectId{resource.kind, obj.Namespace, obj.Name}] = struct{}{}
+			objectsToGarbageCollect[objectId{resource.kind, obj.Namespace, obj.Name}] = &obj
 		}
 	}
 
